Adds nicotine effects to smokables

Kasuobes opened the pull request:
Moved nicotine reagent from toxins to medicine, added new attributes. Nicotine will metabolize at 0.6u per 300 seconds (ie. smoke time for cigarettes)

Added nicotine to cigars and cigarettes, increased chem volume to compensate. One cigarette will pass on 1u of nicotine via ingestion.

mwerezak commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Medicine.dm:
```diff
@@ -510,6 +510,30 @@
+	M.add_chemical_effect(CE_PULSE, 1)
+	if(volume <= 0.15 && data != -1)
+		data = -1
+		M << "<span class='warning'>You grow antsy, craving for another puff of smoke...</span>"
```
I feel like this message should be made more general, since it will apply for injections as well.

mwerezak commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Medicine.dm:
```diff
@@ -510,6 +510,30 @@
+	if(alien == IS_DIONA)
+		return
+	M.add_chemical_effect(CE_PULSE, 1)
+	if(volume <= 0.15 && data != -1)
```
It may be possible to achieve this with a combination of `dose` and `volume`. If so you should probably do that.

Kasuobes commented:
Generalized nicotine withdrawal message, reworked withdrawal message system. to use `volume`, `dose` and `world.time`.

mwerezak commented on code/game/objects/items/weapons/cigs_lighters.dm:
```diff
@@ -287,7 +295,11 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 	icon_off = "cigar2off"
 	smoketime = 7200
-	chem_volume = 30
+	chem_volume = 135
```
This cigar can hold more than a bucket's worth of reagent. This is excessive.

Kasuobes commented:
Decreased chem_volume of cigarettes and cigars, decreased starting amount of nicotine. Adjusted metabolism rate of nicotine to compensate.

Also adjusted cigar and cigarette boxes to reflect the decreased chem_volume.

Kasuobes commented:
Adjusted traitor cigarette boxes starting reagent amounts.

mwerezak commented:
:+1:
